src/apps/external_executor/external_executor.cpp

```diff
@@ -28,32 +28,35 @@ namespace externalexecutor
         auto records_handle = ctx.tx.template rw<Map>(payload.table());
         records_handle->put(payload.key(), payload.value());
 
-        ctx.rpc_ctx->set_response_status(HTTP_STATUS_OK);
+        return ccf::grpc::make_success();
       };
 
       make_endpoint(
         "ccf.KV/Put",
         HTTP_POST,
-        ccf::grpc_adapter<ccf::KVKeyValue, void>(put),
+        ccf::grpc_adapter<ccf::KVKeyValue, ccf::grpc::EmptyResponse>(put),
         ccf::no_auth_required)
         .install();
 
       auto get = [this](
                    ccf::endpoints::ReadOnlyEndpointContext& ctx,
-                   ccf::KVKey&& payload) {
+                   ccf::KVKey&& payload)
+        -> ccf::grpc::GrpcAdapterResponse<ccf::KVValue> {
         auto records_handle = ctx.tx.template ro<Map>(payload.table());
         auto value = records_handle->get(payload.key());
         if (!value.has_value())
         {
-          ctx.rpc_ctx->set_response_status(HTTP_STATUS_NOT_FOUND);
-          return ccf::KVValue(); // Handle errors
+          // Note: no need to specify `make_error<ccf::KVValue>` here as lambda
+          // returns `-> ccf::grpc::GrpcAdapterResponse<ccf::KVValue>`
+          return ccf::grpc::make_error(
```

Comment:
As discussed with @eddyashton, we want to avoid having to specify `make_error<ccf::KVValue>` here as the error doesn't contain any `ccf::KVValue` although the handler is strongly typed (`ccf::grpc_read_only_adapter<ccf::KVKey, ccf::KVValue>(get)`). The hint on line `44` allows us to do that (along with some `std::variant>` shenanigans in `grpc.h`).